### PR TITLE
jobs: use k8s-infra image for wg-k8s-infra jobs

### PR DESCRIPTION
**Pull request by spiffxp**
Now that https://github.com/kubernetes/k8s.io/pull/2134 has merged

The job added in https://github.com/kubernetes/test-infra/pull/22458 has built `gcr.io/k8s-staging-infra-tools/k8s-infra:latest`

Use it for all wg-k8s-infra jobs

### CODE REVIEW

**Comment by spiffxp** [comment]
/hold
Well actually, https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-k8sio-push-image-k8s-infra/1402003668366004224 failed with 
https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-k8sio-push-image-k8s-infra/1402003668366004224#1:build-log.txt%3A13
```
ERROR: (gcloud.builds.submit) interpreting /home/prow/go/src/github.com/kubernetes/k8s.io/images/k8s-infra/cloudbuild.yaml as build config: .options.machineType: unused
```

File: config/jobs/kubernetes/wg-k8s-infra/k8sio-presubmit.yaml
Comment (BenTheElder) [inline config/jobs/kubernetes/wg-k8s-infra/k8sio-presubmit.yaml:16]: 🎉

**Review by BenTheElder — APPROVED** [review]
LGTM

**Comment by spiffxp** [comment]
https://github.com/kubernetes/test-infra/pull/22464 should unblock getting the image pushed

**Comment by spiffxp** [comment]
pull-test-infra-yamllint wasn't happy, force-pushed to fix
```
config/jobs/kubernetes/wg-k8s-infra/k8sio-presubmit.yaml
  18:13     error    trailing spaces  (trailing-spaces)
```

**Comment by spiffxp** [comment]
https://github.com/kubernetes/test-infra/pull/22466 will update the image-pushing job to use an image with the latest version of `gcloud`, one more PR will be needed to trigger a new job to actually push the `k8s-infra:latest` image

**Comment by spiffxp** [comment]
Opened https://github.com/kubernetes/k8s.io/pull/2152 to trigger a new job

**Comment by spiffxp** [comment]
/hold cancel
https://github.com/kubernetes/k8s.io/pull/2154 finally caused an image to get pushed, so, now to verify it works with all of the jobs being updated here

**Pull request merged**
